OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/threading/OneSignalDispatchers.kt

```diff
@@ -87,6 +89,7 @@ internal object OneSignalDispatchers {
                 TimeUnit.SECONDS,
                 LinkedBlockingQueue(QUEUE_CAPACITY),
                 OptimizedThreadFactory(DEFAULT_THREAD_NAME_PREFIX),
+                ThreadPoolExecutor.CallerRunsPolicy(), // Execute on calling thread if queue is full (prevents rejection)
```

Comment:
This can lead to calling things on the main thread, which in turn will lead to ANRs. Also this creates more inconsistencies around timing between thread, which could lead to very hard to reproduce race conditions.

I would rather we continue adding and growing the queue and not reject.